tests/nn/data_parallel/test_fsdp_memory.py

```diff
@@ -236,11 +271,80 @@ def test_fsdp_memory(fsdp, ckpt):
             "iter 2: after step": 66,
             "iter 2: done": 3,
         },
+        ("fsdp_amp_default", "ckpt"): {
+            "iter 0: start": 105,
+            "iter 0: after fwd": 130,
+            "iter 0: after loss": 130,
+            "iter 0: after bwd": 259,
+            "iter 0: after step": 259,
+            "iter 0: done": 105,
+            "iter 1: start": 105,
+            "iter 1: after fwd": 130,
+            "iter 1: after loss": 130,
+            "iter 1: after bwd": 259,
+            "iter 1: after step": 259,
+            "iter 1: done": 105,
+            "iter 2: start": 105,
+            "iter 2: after fwd": 130,
+            "iter 2: after loss": 130,
+            "iter 2: after bwd": 259,
+            "iter 2: after step": 259,
+            "iter 2: done": 105,
+        },
+        ("fsdp_amp_compute_dtype32", "ckpt"): {
+            "iter 0: start": 105,
+            "iter 0: after fwd": 130,
+            "iter 0: after loss": 130,
+            "iter 0: after bwd": 272,
+            "iter 0: after step": 272,
+            "iter 0: done": 105,
+            "iter 1: start": 105,
+            "iter 1: after fwd": 130,
+            "iter 1: after loss": 130,
+            "iter 1: after bwd": 272,
+            "iter 1: after step": 272,
+            "iter 1: done": 105,
+            "iter 2: start": 105,
+            "iter 2: after fwd": 130,
+            "iter 2: after loss": 130,
+            "iter 2: after bwd": 272,
+            "iter 2: after step": 272,
+            "iter 2: done": 105,
+        },
     }[(fsdp, ckpt)]
-    fsdp = fsdp == "fsdp"
-    ckpt = ckpt == "ckpt"
+
+    # Compute the FSDP config.
+    fsdp_config = {}
+
+    # Set mixed precision.
+    if "amp" in fsdp:
+        fsdp_config["mixed_precision"] = True
+
+    # When compute_dtype is FP32, make sure we use clear_autocast_cache.
+    # Setting fp32_reduce_scatter and verbose for more code coverage.
+    if "compute_dtype32" in fsdp:
+        fsdp_config["compute_dtype"] = torch.float32
+        fsdp_config["fp32_reduce_scatter"] = True
+        fsdp_config["clear_autocast_cache"] = True
+        fsdp_config["verbose"] = True
+
+    # Using bigger hidden dimension for AMP to increase the model size
+    # so that bug in handling params will show up but we don't do that
+    # in the base case to keep the test fast.
+    #   hidden_dim 128: model size 4MB
```

Comment:
extra tab?

Reply:
I added the indentation for make it easier to read, like a bullet point list for two different cases.

Reply:
my suggestion is to add a number or "-"/"*" to make that more obvious.

Reply:
sure. done.